netket/models/autoreg.py

```diff
@@ -46,17 +47,32 @@ def __post_init__(self):
             raise ValueError("Only unconstrained Hilbert spaces are supported by ARNN.")
 
     @abc.abstractmethod
-    def conditionals(self, inputs: Array, cache: PyTree) -> Tuple[Array, PyTree]:
+    def _conditional(self, inputs: Array, index: int) -> Array:
         """
-        Computes the probabilities for each spin to take each value.
+        Computes the conditional probabilities for a site to take each value.
+        This method gives the expected output only if the correct caches are given in `variables`.
+        Typically, it should only be called successively with indices 0, 1, 2, ...,
+        as in the autoregressive sampling procedure.
 
         Args:
           inputs: configurations with dimensions (batch, Hilbert.size).
-          cache: auxiliary states, e.g., used to implement fast autoregressive sampling.
+          index: index of the site.
 
         Returns:
-          p: the probabilities with dimensions (batch, Hilbert.size, Hilbert.local_size).
-          cache: the updated cache.
+          The probabilities with dimensions (batch, Hilbert.local_size).
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def conditionals(self, inputs: Array) -> Array:
+        """
+        Computes the conditional probabilities for each site to take each value.
```

Comment:
Can you give an example or better exapain what this does?

Reply:
Do you think it's clear to add this example?
```
Examples:

  >>> p = model.apply(variables, σ, method=model.conditionals)
  >>> print(p[2, 3, :])
  [0.3 0.7]
  # For the 3rd spin of the 2nd sample in the batch,
  # it takes probability 0.3 to be spin down (local state index 0),
  # and probability 0.7 to be spin up (local state index 1).
```

Reply:
that's good